crates/bevy_render/src/render_resource/uniform_buffer.rs

```diff
@@ -8,6 +8,21 @@ use encase::{
 };
 use wgpu::{util::BufferInitDescriptor, BindingResource, BufferBinding, BufferUsages};
 
+/// Stores data to be transferred to the GPU and made accessible to shaders as a uniform buffer.
+///
+/// Uniform buffers are available to shaders on a read-only basis. Uniform buffers are commonly used to make available to shaders
+/// parameters that are constant during shader execution, and are best used for data that is relatively small in size. For
+/// larger data, or data that must be made accessible to shaders on a read-write basis, consider
+/// [`StorageBuffer`](crate::render_resource::StorageBuffer).
+///
+/// The contained data is stored in system RAM. [`write_buffer`](crate::render_resource::UniformBuffer::write_buffer) queues
+/// copying of the data from system RAM to VRAM. Data in uniform buffers must follow [std140 alignment/padding requirements],
+/// which is automatically enforced by this structure. Per the WGPU spec, uniform buffers cannot store runtime-sized array (vectors), or structures with fields that are vectors.
+/// If this is required, consider [`DynamicUniformBuffer`](crate::render_resource::DynamicUniformBuffer).
```

Comment:
There is a bit of a hack one can do to fake runtime sized arrays but I suppose it should be documented in an advanced section of the book rather than here.

Reply:
Okay, so I will leave this section as is. I know you have described this hack to me on the Discord, and it makes sense! I will leave a link here: https://discord.com/channels/691052431525675048/866787577687310356/991373107065532538

And I will also leave a link in the rendering documentation thread on Discord. (see: https://discord.com/channels/691052431525675048/968333504838524958)